tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.utils import (
+    TimeoutSampler, get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    The function has some pre-checks before
+    the openshift-logging upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+def check_csv_logging_upgrade_phase():
```

Comment:
What happen if upgrade did not start at all? i dont see any version validation here - can status be "succeeded" before upgrade even starts?
also, it seems that this function is more general, checking just the status of CSV, regardless if upgrade is triggered or not. function  and variables name should reflect that - my suggestion here is to rename this function and call it every time when this check needed

Reply:
I already check for version before and after upgrade and also check for the Succeeded phase after the upgrade. Also using timeout sampler to check versions and phase after upgrade. Hope this checks are enough. Please let me know.
check_version_after_upgrade(): -----Will have updated CSV after upgrade.